### PR TITLE
Fix for weird color shifting of teleportation target in Rust domain

### PR DESCRIPTION
**Pull request by Zvork**
This PR fixes a bug in the _Haze_ computation when using _Glare_ that had the consequence of applying incorrect colors to some transparent elements such as the teleportation target in the "Rust" domain. The bug was direction dependent (see screenshots for examples of bug).
teleporta
teleportb

# Test Plan
Users should use the corrected _tests/engine/render/effect/haze_ test proposed in [hifi_tests Haze2](https://github.com/highfidelity/hifi_tests/tree/master/tests/engine/render/effect/haze) which takes into account this bug. Checking on the Rust domain (or any domain with a heavy haze with a glare with a color radically different from the haze) is also interesting, by using teleportation in every direction.

The opacity material tests in _tests/engine/render/effect/material/opacity_ should also be run.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8740/app_12873_PR_2adac241d0c647ae1caae1409ac1f455baccf456.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8740/HighFidelity-Beta-PR12873-2adac241d0c647ae1caae1409ac1f455baccf456.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8740/HighFidelity-Beta-Interface-PR12873-2adac241d0c647ae1caae1409ac1f455baccf456.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8740/HighFidelity-Beta-PR12873-2adac241d0c647ae1caae1409ac1f455baccf456.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8740/HighFidelity-Beta-Interface-PR12873-2adac241d0c647ae1caae1409ac1f455baccf456.exe

File: libraries/render-utils/src/Haze.slh [hunk 1]
```diff
@@ -93,25 +93,26 @@ vec3 computeHazeColorKeyLightAttenuation(vec3 color, vec3 lightDirection, vec3 w
...
-    float distance = length(eyeFragPos);
+    float distance = length(fragEyePos);
+    float eyeWorldHeight = eyeWorldPos.y;
 
     // Convert haze colour from uniform into a vec4
     vec4 hazeColor  = vec4(hazeParams.hazeColor, 1.0);
 
     // Directional light component is a function of the angle from the eye, between the fragment and the sun
-    vec3 eyeFragDir = normalize(worldFragPos);
+    vec3 fragToEyeWorldDir = normalize(fragWorldPos-eyeWorldPos);
```
Comment (SamGondelman) [inline libraries/render-utils/src/Haze.slh]: Spaces around `-`.  Looks good otherwise!

**Comment by Zvork** [comment]
There was some more shader errors with translucent errors (eye space / world space mix up) that has been fixed. But as a side effect, the teleport target now has some clearly visible specular highlights... not sure this is wanted, or maybe this is another bug (still investigating).

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8779/HighFidelity-Beta-PR12873-8ef25301cbc0e20419d2f9112a603e15b62f21ee.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8779/HighFidelity-Beta-Interface-PR12873-8ef25301cbc0e20419d2f9112a603e15b62f21ee.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8779/HighFidelity-Beta-PR12873-8ef25301cbc0e20419d2f9112a603e15b62f21ee.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8779/HighFidelity-Beta-Interface-PR12873-8ef25301cbc0e20419d2f9112a603e15b62f21ee.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8810/app_12873_PR_9ad9c070f8e4f83791e3943afe1a3b4f08ee6bc9.apk

**Comment by Zvork** [comment]
I confirm that this fix has revealed a "content bug" on the teleportation target which, with a roughness value of 1.0 on its material, now displays a visible large specular highlight when the light hits it front face. The bug is less visible with a roughness of 0, but still present at grazing angles. The explanation is that opacity doesn't apply to specular and models, due to the PBR shader, always have some sort of specular.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8810/HighFidelity-Beta-PR12873-9ad9c070f8e4f83791e3943afe1a3b4f08ee6bc9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8810/HighFidelity-Beta-Interface-PR12873-9ad9c070f8e4f83791e3943afe1a3b4f08ee6bc9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8810/HighFidelity-Beta-PR12873-9ad9c070f8e4f83791e3943afe1a3b4f08ee6bc9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8810/HighFidelity-Beta-Interface-PR12873-9ad9c070f8e4f83791e3943afe1a3b4f08ee6bc9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8811/HighFidelity-Beta-PR12873-6a089be973b240ead39692c073896540ecc4521a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8811/HighFidelity-Beta-Interface-PR12873-6a089be973b240ead39692c073896540ecc4521a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8811/HighFidelity-Beta-PR12873-6a089be973b240ead39692c073896540ecc4521a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8811/HighFidelity-Beta-Interface-PR12873-6a089be973b240ead39692c073896540ecc4521a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8937/app_12873_PR_c8bef8651f904343901ff704ec4bbd8446a9435b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8937/HighFidelity-Beta-PR12873-c8bef8651f904343901ff704ec4bbd8446a9435b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8937/HighFidelity-Beta-Interface-PR12873-c8bef8651f904343901ff704ec4bbd8446a9435b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8937/HighFidelity-Beta-PR12873-c8bef8651f904343901ff704ec4bbd8446a9435b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8937/HighFidelity-Beta-Interface-PR12873-c8bef8651f904343901ff704ec4bbd8446a9435b.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Teleportation "laser" always appears the correct color, tested in rust-experiments.

**Pull request merged**